multiplayer/src/components/Presence.tsx

```diff
@@ -6,6 +6,27 @@ import Popup from "./Popup";
 import { Button } from "react-common/components/controls/Button";
 import { showModal } from "../state/actions";
 
+function PlayerMenuPopup(
```

Comment:
This approach seems fine, but did you consider trying to memoize it with `useMemo`?

Reply:
I didn't, I've only used `useMemo` with computed values / pure functions vs components so I'm not certain I see how it could be applied in this case -- would it be to memoize the `PlayerMenuPopup` function so that the jsx element it returns out can be compared nicely? This component seems to rely on the `setShowPlayerMenu` being current (& `children` which I'm not sure how they would be compared in the memo) so I'm not sure how that would be handled with `useMemo` without getting stale function references after react rerenders

Reply:
Just a thought I had after a quick look. Your way seems more correct.